### ISSUE TITLE
Ng1ViewDeclaration issue with Typescript 2.4 (NG1 guide might be wrong)

### ISSUE THREAD

**Issue by christophercr** [angular-ui/ui-router#3515]
This issue tracker is for Bug Reports and Feature Requests only.
Please direct requests for help to StackOverflow.  See http://bit.ly/UIR-SOF for details.

This is a:

- [X] Bug Report
- [ ] Feature Request
- [ ] General Query 

My version of UI-Router is: 1.0.0-beta.3

# Bug Report

#### Current Behavior:

So far I've been declaring my states like this with no issues so my app is working correctly:

```javascript
let myState = {
  name: stateName,
  url: stateUrl,
  views: {
    "myParentView@": "myComponent"
  }
}
```

which is the same way illustrated in these docs:
https://ui-router.github.io/guide/views#defining-named-views 
https://ui-router.github.io/ng1/docs/latest/interfaces/ng1.ng1statedeclaration.html#views
https://ui-router.github.io/guide/ng1/route-to-component#named-viewsview-targeting

However, after upgrading Typescript to 2.4, I get compilation issues due to the Ng1ViewDeclaration type: 

`Property '"myParentView@"' is incompatible with index signature. Type 'string' has no properties in common with type 'Ng1ViewDeclaration'.`

In order to fix this, I have to change my state declaration to add the view "component" property:

```javascript
let myState = {
  name: stateName,
  url: stateUrl,
  views: {
    "myParentView@": { component: "myComponent" }
  }
}
```

Although I'm using the 1.0.0-beta.3 version, I suspect the same issue remains in 1.0.x since the Ng1ViewDeclaration types did not change that much.

#### Expected Behavior:

No compilation issues with component name in views object.

**Comment by christophercr** [comment]
Hi, any news about this?

**Comment by christophercr** [comment]
The issue has been solved in [https://github.com/angular-ui/ui-router/commit/2f7a3f2](url) and released in 1.0.7.

Thanks!
